### PR TITLE
Add missing service documentation, move helpers out of transformer file

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR
- Adds missing service documentation, including resource types
- Removes some unnecessary component docs
- Moves some healthcare-service helpers from the transformers file into the index

## Testing done
Unit testing

## Acceptance criteria
- [ ] Build passes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/services/healthcare-service/index.js [hunk 1]
```diff
@@ -65,3 +65,45 @@ export async function getSupportedHealthcareServicesAndLocations({
 
   return results.filter(item => item.resourceType === 'Location');
 }
+
+/**
```
Comment (jbalboni) [inline src/applications/vaos/services/healthcare-service/index.js:69]: This code was moved from the transformers file. No changes except to JSDoc comments

**Review by LeKeve — COMMENTED** [review]
I was also thinking that we could remove all component definitions in this ticket. The ones in `/components/calendar` and `/components/NewTabAnchor`.

**Comment by LeKeve** [comment]
@jbalboni Most of the shared components are pretty small and documenting them seemed like overkill to me. I think the param definitions could be helpful, but having prop types would have a similar benefit and also potentially catch any missing prop issues.

I also don't have a strong opinion either way though. If the team finds the shared component docs useful then we can probably create a ticket to go through them and add the missing docs.

**Review by LeKeve — APPROVED** [review]
Forgot to approve!

**Comment by jbalboni** [comment]
Cool, I'll merge this and ask the team next week. If we need to remove the component docs, I can do that in a follow up PR.

**Pull request merged**